src/pubsub/ua_pubsub_reader.c

```diff
@@ -518,6 +517,34 @@ UA_Server_DataSetReader_getState(UA_Server *server, UA_NodeId dataSetReaderIdent
     return res;
 }
 
+UA_StatusCode
+UA_Server_enableDataSetReader(UA_Server *server, const UA_NodeId dataSetReaderIdentifier) {
+    UA_LOCK(&server->serviceMutex);
+    UA_StatusCode ret = UA_STATUSCODE_BADNOTFOUND;
+    UA_DataSetReader* dsr = UA_ReaderGroup_findDSRbyId(server, dataSetReaderIdentifier);
+
+    if(dsr)
```

Comment:
We prefer early-returns to deeply nested if-else.

Reply:
Thanks, I have updated it to remove one layer of nested if